### DISCUSSION TITLE
Whats the current recommendations for including tables in typst?

### DISCUSSION THREAD

**Discussion by aghaynes** [quarto-dev/quarto-cli#8835] [Q&A]
### Description

When using the typst format, tables currently do not look overly nice

```
mtcars |> head() |> knitr::kable()
```

produces a table that looks like this 
![image](https://github.com/quarto-dev/quarto-cli/assets/6594028/0b5e77a2-ea3c-4afe-922f-e5c029d1eeee)

gt tables are basically the same 
![image](https://github.com/quarto-dev/quarto-cli/assets/6594028/b76cb438-bd69-4b65-975a-e18950fbc1ed)

What is the currently recommended R package for creating nice tables in typst (if any)? Is this something that will be integrated into gt at some point perhaps?

**Comment by mcanouil** [comment]
Not really a Quarto question.

Typst is quite new.
Pandoc does not support all Typst features and does not support all markdown table syntax conversion to Typst.
Since Quarto is using Pandoc for most of the conversion from markdown to formats.

Now, for feature request about adding Typst as output, Search and open a new one if it does not exist:
- https://github.com/yihui/knitr
- https://github.com/rstudio/gt

Regarding R packages, I'm afraid a good search is the best way: https://cran.r-project.org/search.html (and/or on GitHub/GitLab).

**Comment by cderv** [comment]
Just mentioning here for any future reader: 

Currently, Tables will usually be emitted as Markdown table by Tables generation tooling and then converted to Typst format with the Pandoc conversion. Quarto brings support for HTML parsing also, with the same conversion by Pandoc tables. 

We are aware of specific table tools in the Typst ecosystem that could improve the output. This could be a feature request into Pandoc to improve their Typst writer for table output, but we are considering this ourselves. The issue to track for example: 

* https://github.com/quarto-dev/quarto-cli/issues/6811

Also, for reference, this topic has been reported in knitr: 
* https://github.com/yihui/knitr/issues/2328

And for the current existing  feature request I know there is one in **gt**

* https://github.com/rstudio/gt/issues/1500

As shared by @aghaynes on another topic, it seems this R package already has some options to output Typst table: https://github.com/vincentarelbundock/tinytable


Hope it helps
